src/main/java/com/puppycrawl/tools/checkstyle/utils/JavadocUtil.java

```diff
@@ -408,4 +408,47 @@ else if (sibling.getType() == TokenTypes.SINGLE_LINE_COMMENT) {
                 || BlockCommentPosition.isOnPackage(blockComment));
     }
 
+    /**
+     * Gets the content of inline custom tag.
+     *
+     * @param inlineTag inline tag node.
+     * @return String consisting of the content of inline custom tag.
+     */
+    public static String getContentOfInlineCustomTag(DetailNode inlineTag) {
+        final DetailNode[] childrenOfInlineTag = inlineTag.getChildren();
+        final StringBuilder customTagContent = new StringBuilder(256);
+        final int numOfChildrenEmptyInlineTag = 3;
+        if (childrenOfInlineTag.length != numOfChildrenEmptyInlineTag) {
+            final DetailNode endingNode =
+                    childrenOfInlineTag[childrenOfInlineTag.length - 1];
```

Comment:
it looks a bit weird. Why cant it just iterate from specific index until end of array instead of comparing nodes in line 426?

other option is to make while cycle until next sibling is null

Reply:
`null` won't work as we will end up getting the content of `JavadocTokenTypes.JAVADOC_INLINE_TAG_END` i.e. `}`. So instead of getting something like-
```
This is the summary.
```
We will get-
```
This is the summary.}
```
There are various methods of getting around this like-
```
while (currentNode.getType() != JavadocTokenTypes.JAVADOC_INLINE_TAG_END) {
                extractInlineTagContent(currentNode, customTagContent);
                currentNode = getNextSibling(currentNode);
}
```
You could add 
`currentNode.getType() != JavadocTokenTypes.JAVADOC_INLINE_TAG_END` flag anywhere you wish, or we could go with what it is right now.

> it looks a bit weird.

The number play? It is kind of required, we have to start from the `DESCRIPTION` node, so 3 is for that and the minimum no of children of `JAVADOC_INLINE_TAG` is 4. If the no of children is equal to 3 then it is an empty summary i.e. `{@summary}`.

```
|--JAVADOC_INLINE_TAG -> JAVADOC_INLINE_TAG [1:3]
|   |--JAVADOC_INLINE_TAG_START -> { [1:3]
|   |--CUSTOM_NAME -> @summary [1:4]
|   |--WS ->   [1:12]
|   |--DESCRIPTION -> DESCRIPTION [1:13]
|   |   |--TEXT -> <!-- This is a HTML comment --> [1:13]
|   |   |--NEWLINE -> \n [1:44]
|   |   |--LEADING_ASTERISK ->  * [2:0]
|   |   |--TEXT ->   <ul> [2:2]
|   |   |--NEWLINE -> \n [2:8]
|   |   |--LEADING_ASTERISK ->  * [3:0]
|   |   |--TEXT ->    <li>Coffee</li> [3:2]
|   |   |--NEWLINE -> \n [3:20]
|   |   |--LEADING_ASTERISK ->  * [4:0]
|   |   |--TEXT ->  </ul> [4:2]
|   |   |--NEWLINE -> \n [4:8]
|   |   |--LEADING_ASTERISK ->  * [5:0]
|   |   `--TEXT ->  **** some stars [5:2]
|   `--JAVADOC_INLINE_TAG_END -> } [5:18]

```

Reply:
> currentNode.getType() != JavadocTokenTypes.JAVADOC_INLINE_TAG_END

this seems better to me

Reply:
Done.